app/routes/v2/webhooks.js

```diff
@@ -0,0 +1,132 @@
+
+router.use(ebl(getBunyanConfig('razeedash-api/webhooks')));
+
+// Callback from triggered webhook
+const addCallbackResult = async (req, res, next) => {
+  try {
+    const Webhooks = req.db.collection('webhooks');
+    const webhook = await Webhooks.findOne({ _id: req.params.webhook_id });
+    if (webhook) {
+      // Make sure webhook is still active
+      if (webhook.deleted == true) {
+        res.status(404).send('Web hook has been deleted');
+      } else {
+        // Validate badge properties
+        let badge = req.body;
+        badge.webhook_id = req.params.webhook_id;
+        let properties = ['webhook_id', 'url', 'description', 'link', 'status'];
+        let isValid = jkValidate(badge, properties);
+        if ((webhook.trigger == WEBHOOK_TRIGGER_IMAGE) && (!badge.image_id)) {
+          properties.push('image_id');
+          isValid = false;
+        }
+        if (!isValid) {
+          res.status(400).send(`Missing properties, make sure the following fields are defined: ${JSON.stringify(properties)}`);
+        } else {
+          // determine resource to add badge
+          // if resource not currently used return 404
+          if (webhook.trigger == WEBHOOK_TRIGGER_CLUSTER) {
+            const cluster = await insertClusterBadge(webhook, badge, req);
+            if (cluster) {
+              res.status(201);
+            } else { // should never happen
+              res.log.error({ badge: badge, webhook: webhook }, 'cluster missing while processing badge ');
+              res.status(500);
+            }
+          } else if (webhook.trigger == WEBHOOK_TRIGGER_IMAGE) {
+            const image = await insertImageBadge(badge, req);
+            if (image) {
+              res.status(201);
+            } else { // should never happen
+              res.log.error({ badge: badge, webhook: webhook }, 'image missing while processing badge ');
+              res.status(500);
+            }
+          } else { // should never happen
+            res.log.error({ badge: badge, webhook: webhook }, 'Unknown webhook trigger defined in database');
+            res.status(500).send('unknown trigger');
+          }
+        }
+      }
+    } else {
+      res.status(404).send('Web hook not found');
+    }
+  } catch (err) {
+    req.log.error(err);
+    next(err);
+  }
+};
+
+const addWebhook = async (req, res, next) => {
+  try {
+    let webhook = req.body;
+    const Webhooks = req.db.collection('webhooks');
+    if (webhook.trigger == WEBHOOK_TRIGGER_CLUSTER) {
+      const Clusters = req.db.collection('clusters');
+      const result = await Clusters.findOne({ cluster_id: webhook.cluster_id, org_id: req.org._id });
+      if ((!result) || (result && result.deleted == true)) {
+        res.status(404).send(`Cluster ${webhook.cluster_id} not found or has been deleted`);
+      } else {
+        await Webhooks.insertOne(webhook);
```

Comment:
We should add the ord_id to the webhook document to facilitate future cleanup when an org is deleted.

Reply:
done